File: frontend/src/api/project.ts
import api from './index'

export interface Project {
  id: number
  name: string
  description: string
  prometheus_url: string
  token?: string
  created_at: string
  updated_at: string
}

export const projectApi = {
  list: () => api.get<any, { data: Project[] }>('/projects'),

  get: (id: number) => api.get<any, { data: Project }>(`/projects/${id}`),

  create: (data: Partial<Project>) => api.post<any, { data: Project }>('/projects', data),

  update: (id: number, data: Partial<Project>) => api.put<any, { data: Project }>(`/projects/${id}`, data),

  delete: (id: number) => api.delete(`/projects/${id}`)
}

File: backend/internal/model/project.go
package model

import (
	"time"
)

// Project 项目表
type Project struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	Name          string    `gorm:"uniqueIndex;size:100;not null" json:"name"`
	Description   string    `gorm:"size:500" json:"description"`
	PrometheusURL string    `gorm:"size:255;not null" json:"prometheus_url"`
	Token         string    `gorm:"size:255" json:"token,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

func (Project) TableName() string {
	return "projects"
}
